docs/api/media.rst

```diff
@@ -0,0 +1,110 @@
+.. _media:
+
+Media
+=====
+
+Falcon allows for easy and customizable internet media type handling. By default
+Falcon only enables a single JSON handler. However, additional handlers
+can be configured through the :any:`falcon.RequestOptions` and
+:any:`falcon.ResponseOptions` objects specified on your :any:`falcon.API`.
+
+.. note::
+
+    To avoid unnecessary overhead, Falcon will only process request media
+    the first time the media property is referenced. Once it has been
+    referenced, it'll use the cached result for subsequent interactions.
+
+Usage
+-----
+
+Zero configuration is needed if you're creating a JSON API. Just access
+or set the ``media`` attribute as appropriate and let Falcon do the heavy
+lifting for you.
+
+.. code:: python
+
+    import falcon
+
+
+    class EchoResource(object):
+        def on_post(self, req, resp):
+            message = req.media.get('message')
+
+            resp.media = {'message': message}
+            resp.status = falcon.HTTP_200
+
+.. warning::
+
+    Once `media` is called on a request, it'll consume the request's stream.
+
+Validating Media
+----------------
+
+Falcon currently only supports a JSONSchema media type handler; however,
+JSONSchema is very versatile and can be used to validate any deserialized
+media type that JSON also supports (i.e. dicts, lists, etc).
+
+.. autofunction:: falcon.media.validators.jsonschema.validate
+
+
+Content-Type Negotiation
+------------------------
+
+Falcon currently only supports partial negotiation out of the box. By default,
+when the ``media`` attribute is used it attempts to de/serialize based on the
+``Content-Type`` header value. The missing link that Falcon doesn't provide
+is the connection between the :any:`falcon.Request` ``Accept`` header provided
+by a user and the :any:`falcon.Response` ``Content-Type`` header.
+
```

Comment:
[suggestion] We should probably say something about why this isn't the end of the world. Let's not end on a negative note.